src/main/java/ch/cern/eam/wshub/core/tools/FieldDescriptionTools.java

```diff
@@ -137,6 +138,24 @@ public String readCostCodeDesc(InforContext inforContext, String costCode) {
 		return getDescription(inforContext, gridRequest, "des_text");
 	}
 
+	public String readTypeDesc(InforContext inforContext, String entity, String typeCode) {
+		if (isEmpty(typeCode)) return null;
+		GridRequest gridRequest = new GridRequest("BSUCOD_HDR", GridRequest.GRIDTYPE.LOV, 1);
+		gridRequest.addParam("param.entitycode", entity);
+		gridRequest.setUserFunctionName("BSUCOD");
+		gridRequest.addFilter("usercode", typeCode, "EQUALS");
+		return getDescription(inforContext, gridRequest, "usercodedescription");
+	}
+
+	public String readStatusDesc(InforContext inforContext, String entity, String statusCode) {
+		if (isEmpty(statusCode)) return null;
+		GridRequest gridRequest = new GridRequest("BSUCOD_HDR", GridRequest.GRIDTYPE.LOV, 1);
+		gridRequest.addParam("param.entitycode", entity);
+		gridRequest.setUserFunctionName("BSUCOD");
+		gridRequest.addFilter("usercode", statusCode, "EQUALS");
+		return getDescription(inforContext, gridRequest, "usercodedescription");
+	}
+
```

Comment:
Hey Miguel, what do you think about merging those two methods into one - public String readUserCodeDesc(InforContext inforContext, String entityCode, String userCode)?

Reply:
Absolutely!